legacy/reqcounter/reqcounter_test.go

```diff
@@ -124,3 +126,48 @@ func TestRequestCounterIncrement(t *testing.T) {
 	// Ensure the request counter was incremented.
 	require.EqualValues(t, &expected, &requestCounter, "The request counter failed to increment its request field.")
 }
+
+func TestLog(t *testing.T) {
+	// Create multiple request counters with unique intervals.
+	requestCounters := []rc.RequestCounter{
+		NewRequestCounter(0),
+		NewRequestCounter(9),
+		NewRequestCounter(2839),
+	}
+	requestCounters[0].Interval = time.Second * 2
+	requestCounters[1].Interval = time.Second
+	requestCounters[2].Interval = time.Hour
+	// Load request counters into network monitor.
+	netMonitor := &rc.NetworkMonitor{
+		RequestCounters: rc.RequestCounters{
+			&requestCounters[0],
+			&requestCounters[1],
+			&requestCounters[2],
+		},
+	}
+	// Mock the Debug function, capturing each statement.
+	logged := []string{}
+	rc.Debug = func(args ...interface{}) {
+		logged = append(logged, fmt.Sprint(args[0]))
+	}
+	// Mock the global clock with fake time.
+	fakeTime := &ft.FakeTime{}
+	timeStep := time.Second * 2
```

Comment:
Just to prove that this fake function actually works like we expect it to, can you change this to be a 1 hour interval instead of 2 seconds?

Reply:
This PR is not ready for review. It was only pushed for proof of concept.